app/src/main/java/com/github/sdpteam15/polyevents/view/adapter/EventListAdapter.kt

```diff
@@ -0,0 +1,201 @@
+    }
+
+
+    abstract inner class CustomViewHolder<T>(view: View) :
+            RecyclerView.ViewHolder(view) {
+        abstract fun bind(value: T)
+        abstract fun unbind()
+    }
+
+    /**
+     * Adapted ViewHolder for each item type
+     * Takes the corresponding item type "tab" view
+     */
+    inner class ZoneViewHolder(private val view: View) :
+            CustomViewHolder<String>(view) {
+        private val zoneName = view.findViewById<TextView>(R.id.id_zone_name_text)
+
+        override fun bind(value: String) {
+            zoneName.text = allEvents[value]!!.first
+            view.setOnClickListener {
+                isZoneOpen[value] = !isZoneOpen[value]!!
+                notifyDataSetChanged()
+            }
+        }
+
+        override fun unbind() {
+            //do nothing
+        }
+
+    }
+
+    /**
+     * Adapted ViewHolder for each item
+     * Takes the corresponding item "tab" view
+     */
+    inner class EventViewHolder(private val view: View) :
+            CustomViewHolder<Event>(view) {
+
+        private lateinit var event: Event
+
+        /**
+         * Binds the value of the item to the layout of the item tab
+         */
+        override fun bind(value: Event) {
+            this.event = value
+            view.findViewById<TextView>(R.id.tvEventTitle).text = event.eventName
+            view.findViewById<TextView>(R.id.tvEventStartDate).text = localDatetimeToString(event.startTime, noHourText)
+            view.findViewById<TextView>(R.id.tvEventEndDate).text = localDatetimeToString(event.endTime, noHourText)
+            view.findViewById<ImageButton>(R.id.idEditEventButton).setOnClickListener { modifyListener(event.eventId!!) }
+            view.findViewById<ImageButton>(R.id.idDeleteEventButton).setOnClickListener {
+                deleteListener(event.zoneId!!, event)
+            }
+        }
+
+        override fun unbind() {
+            //Do nothing
+        }
+    }
+
+
+    override fun getItemCount(): Int {
+        var count = 0
+        for (isOpen in isZoneOpen.entries) {
+            count += 1 + (if (!isOpen.value) 0 else allEvents[isOpen.key]?.second?.size ?: 0)
+        }
+        return count
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CustomViewHolder<*> {
+        //Create the correct holder
+        return when (viewType) {
+            ZONE_HOLDER -> {
+                val view = inflater.inflate(R.layout.card_zone_name, parent, false)
+                ZoneViewHolder(view)
+            }
+            EVENT_HOLDER -> {
+                val view = inflater.inflate(R.layout.card_event_by_zone, parent, false)
+                EventViewHolder(view)
+            }
+            //Should not happen
+            else -> throw IllegalArgumentException("wrong view type $viewType")
+        }
+    }
+
+    override fun onViewRecycled(holder: CustomViewHolder<*>) {
+        super.onViewRecycled(holder)
+        holder.unbind()
+    }
```

Comment:
if you delete the unbind, you can remove these lines too
```suggestion
    
```

Reply:
Okay 👍